test/js/bun/test/test-test.test.ts

```diff
@@ -161,6 +161,10 @@ test("testing Bun.deepEquals() using isEqual()", () => {
   expect(Infinity).toEqual(1 / 0);
   expect(-Infinity).toEqual(-Infinity);
   expect(-Infinity).toEqual(-1 / 0);
+
+  expect(Error("foo")).toEqual(Error("foo"));
+  expect(Error("foo")).not.toEqual(Error("bar"));
+  expect(Error("foo")).not.toEqual("foo");
```

Comment:
What happens if the `Error` is a subclass of `Error`? 

```suggestion
  class CustomError extends Error { constructor(message) { super(message); } };
  expect(new CustomError("foo")).not.toEqual(new CustomError("bar"));
  expect(new CustomError("foo")).toEqual(new CustomError("foo"));
```

Does jest behave similar here?

Reply:
Yes, it behaves similarly on Jest. I'll add that to the tests.

![image](https://github.com/oven-sh/bun/assets/11046907/2240a8f8-1377-46a3-a60f-f3b1c205d10d)